app/src/main/java/com/example/androidacademy/api/MovieDtoToDomain.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.androidacademy.api
+
+import com.example.androidacademy.BuildConfig
+import com.example.androidacademy.data.Movie
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+suspend fun convertMovieDtoToDomain(moviesDto: List<MovieDto>, genres: List<GenreDto>): List<Movie> =
+    withContext(Dispatchers.Default) {
+        val genresMap: Map<Int, GenreDto> = genres.associateBy { it.id }
+
+        moviesDto.map { movieDto: MovieDto ->
+            Movie(
+                id = movieDto.id,
+                title = movieDto.title,
+                overview = movieDto?.overview,
+                poster = movieDto.poster.let { BuildConfig.BASE_IMAGE_URL + movieDto.poster },
```

Comment:
Пропущен символ вопроса, перед let.
```suggestion
                poster = movieDto.poster?.let { BuildConfig.BASE_IMAGE_URL + movieDto.poster },
```